src/Core/CFamily/EffectiveRulesConfigCalculator.cs

```diff
@@ -64,32 +64,34 @@ public EffectiveRulesConfigCalculator(ILogger logger)
             configCache = new RulesConfigCache();
         }
 
-        public ICFamilyRulesConfig GetEffectiveRulesConfig(string languageKey, ICFamilyRulesConfig defaultRulesConfig, RulesSettings customRulesConfiguration)
+        public ICFamilyRulesConfig GetEffectiveRulesConfig(string languageKey, ICFamilyRulesConfig defaultRulesConfig, RulesSettings customSettings)
         {
             if (defaultRulesConfig == null)
             {
                 throw new ArgumentNullException(nameof(defaultRulesConfig));
             }
-
-            // Optimisation - if there are no custom rules settings then just return the default
-            if ((customRulesConfiguration?.Rules?.Count ?? 0) == 0)
+            var effectiveConfig = configCache.FindConfig(languageKey, defaultRulesConfig, customSettings);
+            if (effectiveConfig != null)
             {
-                logger.WriteLine(CoreStrings.EffectiveRules_NoCustomRulesSettings);
-                return defaultRulesConfig;
+                logger.WriteLine(CoreStrings.EffectiveRules_CacheHit);
+                return effectiveConfig;
             }
 
-            var effectiveConfig = configCache.FindConfig(languageKey, defaultRulesConfig, customRulesConfiguration);
-            if (effectiveConfig != null)
+            if ((customSettings?.Rules?.Count ?? 0) == 0)
             {
-                logger.WriteLine(CoreStrings.EffectiveRules_CacheHit);
+                logger.WriteLine(CoreStrings.EffectiveRules_NoCustomRulesSettings);
+                customSettings = new RulesSettings();
             }
             else
             {
                 logger.WriteLine(CoreStrings.EffectiveRules_CacheMiss);
-                effectiveConfig = new DynamicCFamilyRulesConfig(defaultRulesConfig, customRulesConfiguration);
-                configCache.Add(languageKey, defaultRulesConfig, customRulesConfiguration, effectiveConfig);
             }
 
+            // TODO: apply filters to exclude which should not be run
+            effectiveConfig = new DynamicCFamilyRulesConfig(defaultRulesConfig, customSettings);
```

Comment:
The purpose of this change is so that all code paths that involve cache misses reach this point.
That gives a us single location in which to apply the project rule exclusions (either in this class or the `DynamicCFamilyRulesConfig` class.